src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -572,6 +581,39 @@ private JsValue IncludeDoc(JsValue self, JsValue[] args)
                 return self;
             }
 
+            private JsValue IncludeCompareExchangeValue(JsValue self, JsValue[] args)
+            {
+                if (args.Length != 1)
+                    throw new InvalidOperationException("includes.cmpxchg(key) must be called with a single argument");
+
+                if (args[0].IsNull() || args[0].IsUndefined())
+                    return self;
+
+                if (args[0].IsArray())// recursive call ourselves
+                {
+                    var array = args[0].AsArray();
+                    foreach (var pair in array.GetOwnProperties())
+                    {
+                        args[0] = pair.Value.Value;
```

Comment:
Don't we need to filter `length` here?

Reply:
Yes! I've introduced an extension method to handle this